scripts/fuzz_opt.py

```diff
@@ -532,18 +559,53 @@ def handle_pair(self, input, before_wasm, after_wasm, opts):
         # trap on many things wasm would, and in those cases it can do weird
         # undefined things. in such a case, at least compare up until before
         # the trap, which lets us compare at least some results in some cases.
-        interpreter_output = run([in_bin('wasm-opt'), before_wasm, '--fuzz-exec-before'])
-        if TRAP_PREFIX in interpreter_output:
-            trap_index = interpreter_output.index(TRAP_PREFIX)
+        # (this is why wasm2js is not in CompareVMs, which does full
+        # comparisons - we need to limit the comparison in a special way here)
+        interpreter = run([in_bin('wasm-opt'), before_wasm_temp, '--fuzz-exec-before'])
+        if TRAP_PREFIX in interpreter:
+            trap_index = interpreter.index(TRAP_PREFIX)
             # we can't test this function, which the trap is in the middle of.
             # erase everything from this function's output and onward, so we
             # only compare the previous trap-free code
-            call_start = interpreter_output.rindex(FUZZ_EXEC_CALL_PREFIX, 0, trap_index)
-            call_end = interpreter_output.index('\n', call_start)
-            call_line = interpreter_output[call_start:call_end]
+            call_start = interpreter.rindex(FUZZ_EXEC_CALL_PREFIX, 0, trap_index)
+            call_end = interpreter.index('\n', call_start)
+            call_line = interpreter[call_start:call_end]
             before = before[:before.index(call_line)]
             after = after[:after.index(call_line)]
-        compare(before, after, 'Wasm2JS')
+            interpreter = interpreter[:interpreter.index(call_line)]
+
+        def fix_output_for_js(x):
+            # start with the normal output fixes that all VMs need
+            x = fix_output(x)
+
+            # check if a number is 0 or a subnormal, which is basically zero
+            def is_basically_zero(x):
+                # to check if something is a subnormal, compare it to the largest one
+                return x >= 0 and x <= 2.22507385850720088902e-308
+
+            def fix_number(x):
+                x = x.group(1)
+                try:
+                    x = float(x)
+                    # There appear to be some cases where JS VMs will print
+                    # subnormals in full detail while other VMs do not, and vice
+                    # versa. Ignore such really tiny numbers.
+                    if is_basically_zero(x):
+                        x = 0
+                except ValueError:
+                    # not a floating-point number, nothing to do
+                    pass
+                return ' => ' + str(x)
+
+            return re.sub(r' => (-?[\d+-.e\-+]+)', fix_number, x)
```

Comment:
What black magic is this!?

Reply:
"simple" regex to detect a subset of floating point numbers... The `=>` is unrelated, I'll clarify that in a comment.